### PR TITLE
Initial Armada Commit

### PR DESCRIPTION
**Pull request by alanmeadows**

### CODE REVIEW

File: armada/chartbuilder.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        We only support a git source type right now, which can also
+        handle git:// local paths as well
+        '''
+
+        if self.chart.source.type == 'git':
+
+            tmpdir = tempfile.mkdtemp(prefix='armada', dir='/tmp')
+            self._source_tmp_dir = tmpdir
+            if self.parent:
+                LOG.info("Cloning %s/%s as dependency for %s" % (self.chart.source.location, self.chart.source.subpath, self.parent))
```
Comment (gardlt) [inline armada/chartbuilder.py]: This is the old way of formatting string. I we should move to the new format `'{} {}'.format('one', 'two')`

File: armada/chartbuilder.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    turning those into proper protoc helm charts that can be
+    pushed to tiller.
+
+    It also processes chart source declarations, fetching chart
+    source from external resources where necessary
+    '''
+
+    def __init__(self, chart, parent=None):
+
+        # cache for generated protoc chart object
```
Comment (gardlt) [inline armada/chartbuilder.py]: if would suggest using a docstring here instead of individual comments.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,91 @@
-# armada
+# Armada
 A python orchestrator for a installing, upgrading, and managing a collection of helm charts, dependencies, and values overrides.
+
+Note that this project is pre-alpha and under active development. It may undergo drastic changes to support the long-term vision but contributions are welcome. 
+
+# Overview
+
+TThe armada python library and command line tool provides a way to synchronize an operators intended state, consisting many several charts, dependencies, and overrides using a single file, with an intended helm target.  This allows operators to define many charts, different namespaces for those releases, and overrides in a central place and with a single command, deploy and upgrade them all where applicable.
```
Comment (technosophos) [inline README.md]: Not that I'm qualified to comment on any of this... but did notice the `TThe` at the front of this sentence.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,91 @@
...
+
+It supports fetching helm chart source from various local and remote locations, such as git/github endpoints.  In the future, it may supprot other mechanisms.
+
+It should also give the operator some indication of what is about to change by assisting with diffs for both values, values overrides, and template changes.
+
+Its functionality may extend beyond helm, assisting in interacting with kubernetes directly to perform basic pre and post steps, such as removing completed or failed jobs, or deleting resources that do not support upgrades.  However, primarily, it will be an interface to support orchestrating Helm.
```
Comment (technosophos) [inline README.md]: Given where Helm is today, I think Armada looks like a very solid approach for handling highly sophisticated applications or suites of applications. Excited to see how this develops!

File: armada/chartbuilder.py (hunk 1 above)
Comment (wilkers-steve) [inline armada/chartbuilder.py]: Agree here, but more of a style nit as both are still acceptable.

File: README.md (hunk 4 above)
Comment (wilkers-steve) [inline README.md]: 👍

**Comment by wilkers-steve** [comment]
This looks like a great start @alanmeadows.  Awesome work!

**Comment by intlabs** [comment]
I like the look of where this is going, and from our discussions really think this could begin to address some of our current and future pinch points. It might be a bit rough around the edges now, but we can get these addressed once we've proven the concept and fleshed out the functionality. Really nice work Alan :)

**Comment by v1k0d3n** [comment]
@alanmeadows a rockstar! :1st_place_medal: who says that all "cool tools" need to be written in go, right @technosophos? ;-)

**Comment by wilkers-steve** [comment]
@v1k0d3n 😐

**Comment by alanmeadows** [comment]
Thanks for the feedback everyone.  Many of your comments were addressed.  I'm going to let this in so we can start building on it.  I hope that no eyes were too badly damaged by the rushed python code between conference calls.

**Pull request merged**